src/components/functional/HeadingWizard.tsx

```diff
@@ -0,0 +1,17 @@
+import { FunctionalComponent, h } from "@stencil/core";
+import { JSXBase } from "@stencil/core/internal";
+
+export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
+
+interface HeadingWizardProps extends JSXBase.HTMLAttributes {
+  level: HeadingLevel;
+}
+
+// (∩｀-´)⊃━☆ﾟ.*･｡ﾟ
+export const HeadingWizard: FunctionalComponent<HeadingWizardProps> = (props, children) => {
+  const HeadingTag = `h${props.level}`;
+
+  delete props.level;
+
+  return <HeadingTag {...props}>{children}</HeadingTag>;
```

Comment:
Should this also have something for non-heading elements that support `aria-level`? https://www.w3.org/TR/html-aria/

Reply:
@jcfranco I don't think we want to support using `role="heading"` unless we have a good use case for that. Seems like we would always just be semantic?

Reply:
@paulcpederson I did find one heading role here: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.tsx#L119-L121

Can we just use a real heading here? Was there a good reason to use the role?

Reply:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role#best_practices